Mathlib/Analysis/Complex/UpperHalfPlane/MoebiusAction.lean

```diff
@@ -41,113 +40,174 @@ scoped notation:1024 "↑ₘ[" R "]" A:1024 =>
   ((A : GL (Fin 2) R) : Matrix (Fin 2) (Fin 2) R)
 
 /-- Numerator of the formula for a fractional linear transformation -/
-def num (g : GL(2, ℝ)⁺) (z : ℍ) : ℂ := g 0 0 * z + g 0 1
+def num (g : GL (Fin 2) ℝ) (z : ℂ) : ℂ := g 0 0 * z + g 0 1
 
 /-- Denominator of the formula for a fractional linear transformation -/
-def denom (g : GL(2, ℝ)⁺) (z : ℍ) : ℂ := g 1 0 * z + g 1 1
+def denom (g : GL (Fin 2) ℝ) (z : ℂ) : ℂ := g 1 0 * z + g 1 1
 
-theorem linear_ne_zero (cd : Fin 2 → ℝ) (z : ℍ) (h : cd ≠ 0) : (cd 0 : ℂ) * z + cd 1 ≠ 0 := by
+theorem linear_ne_zero {cd : Fin 2 → ℝ} {z : ℂ} (hz : z.im ≠ 0) (h : cd ≠ 0) :
```

Comment:
I wonder if we want a version of this for `(z : ℍ)`, since thats the case thats going to be used most often and would save you having to put `z.ne` (or whatever it is) every time.